### PR TITLE
split train bus fields

### PR DESCRIPTION
**Pull request by pjaudiomv**

### CODE REVIEW

**Comment by radius314** [comment]
why does that text even exist?

**Comment by pjaudiomv** [comment]
its weird, has to do with adding custom fields, these fields must of been added later or a part of that. the response looks like this.  so this first part is the field title

`bus_lines:  "Bus Lines#@-@#Redline to havaad"`

**Comment by pjaudiomv** [comment]
https://www.nerna.org/main_server/client_interface/json/?switcher=GetSearchResults&meeting_ids[]=443

**Comment by radius314** [comment]
Wouldn't all plugins have this issue and require the same "field massaging".  What's the point of this at all?

**Comment by pjaudiomv** [comment]
yes they would and they do, bread and the bmlt wordpress satellite deal with it

**Comment by radius314** [comment]
What's is the purpose of this extra data in the output response?

**Comment by pjaudiomv** [comment]
https://github.com/bmlt-enabled/bread/blob/cb8b3b49ddbbe335181a39c76bd41f148dabce96/bmlt-meeting-list.php#L1689
https://github.com/bmlt-enabled/bread/blob/cb8b3b49ddbbe335181a39c76bd41f148dabce96/bmlt-meeting-list.php#L1791

**Comment by pjaudiomv** [comment]
https://github.com/bmlt-enabled/bmlt-satellite-base-class/blob/61f5fbbafaca279f0d5941d73dde18fe5c7efe47/nouveau_map_search.js#L3226

**Comment by radius314** [comment]
I do not want to perpetuate making this change into crouton until I know the point of this.

**Comment by pjaudiomv** [comment]
the reason its there is that you can create custom fields with the bmlt, and the bmlt would have no way of knowing what the field title is. this was implemented as a way to be able to have the field title and value in the same response. all of the fields in the other tab are unique in that way. however bus liens and train lines are only visible ones.

**Comment by pjaudiomv** [comment]
so if I create a field in this table it will automatically show up in the other tab and populate in the semantic responses

Screen Shot 2020-03-17 at 11 01 03 AM

**Comment by radius314** [comment]
Would you use the field name?

**Comment by pjaudiomv** [comment]
anything that gets added with a meetingid_bigint of 0 and visibility set to null will populate and have same response

**Comment by pjaudiomv** [comment]
no just field key

**Comment by radius314** [comment]
Why isn’t that sufficient?  In what cases is it important to have a pretty label?  The root server?

**Comment by radius314** [comment]
Couldn’t there be some cleanup on the key to make it pretty?

**Comment by pjaudiomv** [comment]
im guessing the thought was if someone else consumed the data they would have the field name, and the field key could be whatever so you couldnt always rely to derive the field name from the key.

**Comment by radius314** [comment]
Replace underscores with spaces and then
https://www.w3schools.com/php/func_string_ucwords.asp

**Comment by radius314** [comment]
this bad design is leaking unnecessary change into clients.

**Comment by pjaudiomv** [comment]
that's making an assumption the field key is the same as the name, in the case of trains and busses it is. but contact fields its not

**Comment by radius314** [comment]
I’d approach this a lot different, either as it’s own key value pair in the
response, or just not at all.  There are way more clients that are doing
straight up data consumption versus data input.

**Comment by radius314** [comment]
They should just be the same.

**Comment by pjaudiomv** [comment]
I hear you, that is what it is now though and if the root server got changed at least these consumers would need to be changed too `BMLTSearch, Bread, tomato, wordpress satellite` probably yap legacy data migrations as well

**Comment by radius314** [comment]
:(

File: croutonjs/src/js/crouton-core.js [hunk 1]
```diff
@@ -498,6 +498,14 @@ function Crouton(config) {
 					? meetingData[m]['location_info'].replace('/(http|https):\/\/([A-Za-z0-9\._\-\/\?=&;%,]+)/i', '<a style="text-decoration: underline;" href="$1://$2" target="_blank">$1://$2</a>')
 					: "";
 			meetingData[m]['map_word'] = self.localization.getWord('map').toUpperCase();
+			if (meetingData[m]['train_lines'] != null) {
```
Comment (radius314) [inline croutonjs/src/js/crouton-core.js]: could you look for the presence of the glasses separator `#@-@#` and then handle that way versus binding to specific fields?

Comment (pjaudiomv) [reply croutonjs/src/js/crouton-core.js]: that makes sense, then it would handle custom fields. like boat lines

Comment (radius314) [reply croutonjs/src/js/crouton-core.js]: It should handle `ferry_lines` or `closest_blue_bikes_dock`

File: croutonjs/src/js/crouton-core.js [hunk 2]
```diff
@@ -498,13 +498,13 @@ function Crouton(config) {
...
 					: "";
 			meetingData[m]['map_word'] = self.localization.getWord('map').toUpperCase();
-			if (meetingData[m]['train_lines'] != null) {
-				var trainSplit = meetingData[m]['train_lines'].split('#@-@#');
-				meetingData[m]['train_lines'] = trainSplit[1]
-			}
-			if (meetingData[m]['bus_lines'] != null) {
-				var busSplit = meetingData[m]['bus_lines'].split('#@-@#');
-				meetingData[m]['bus_lines'] = busSplit[1]
+			for (var k in  meetingData[m]) {
```
Comment (radius314) [inline croutonjs/src/js/crouton-core.js]: look like a double space in there.

File: croutonjs/src/js/crouton-core.js [hunk 3]
```diff
@@ -498,13 +498,13 @@ function Crouton(config) {
...
+				if (meetingData[m].hasOwnProperty(k) &&  typeof meetingData[m][k] === 'string') {
```
Comment (radius314) [inline croutonjs/src/js/crouton-core.js]: look like a double space in there.

File: croutonjs/src/js/crouton-core.js (hunk 2 above)
Comment (pjaudiomv) [reply croutonjs/src/js/crouton-core.js]: got it, just fixed and rebased

Comment (pjaudiomv) [reply croutonjs/src/js/crouton-core.js]: shit there were two

Comment (pjaudiomv) [reply croutonjs/src/js/crouton-core.js]: ok got that one too

**Comment by radius314** [comment]
You will need to update the release notes and version.

**Pull request merged**